projects/storefrontlib/src/cms-components/checkout/components/delivery-mode/delivery-mode.component.spec.ts

```diff
@@ -230,8 +230,14 @@ describe('DeliveryModeComponent', () => {
 
     it('should call "next" function after being clicked', () => {
       spyOn(component, 'next');
-      getContinueBtn().nativeElement.click();
+
+      // use below code to enable Continue button again, otherwise
+      // the expect method call will fail
+      component.mode.controls['deliveryModeId'].setValue(
```

Comment:
no idea what this is for? if test, after clicking, is setting `continueButtonPressed` variable to `true`, you should probably use some `beforeEach()` and set `component.continueButtonPressed` to `true` before each test runs. thanks to this, tests will be independent of each other

Reply:
Hi Kacper, let me explain:

I agree that each unit test spec should remain isolated. Taking the spec "should call "next" function after being clicked" for example, the "continue" button in delivery mode is by default disabled because:

* the property binding in html template:
```html
[disabled]="deliveryModeInvalid"
```

* in delivery-mode.component.ts:

```typescript
get deliveryModeInvalid(): boolean {
    return this.mode.controls['deliveryModeId'].invalid;
  }
```

and I wrote simple test code to confirm this:

```typescript
 fixture.detectChanges();
    const button = fixture.debugElement.query(By.css('.cx-checkout-btns .btn-primary'));
    console.log('is continue button disabled? ' + button.nativeElement.disabled);
```

the console.log displays "true".

Back to my test spec, in order to ensure that the continue button COULD BE CLICKED by unit test code, I must enable it first. 

As a result I set a valid delivery mode code MANUALLY by the code below, to prepare it for the next click() call:

```typescript
component.mode.controls['deliveryModeId'].setValue(mockDeliveryMode1.code);
```